src/zeit/content/volume/browser/form.py

```diff
@@ -87,23 +83,38 @@ def setUpWidgets(self, *args, **kw):
             self.widgets['volume'].setRenderedValue(settings.default_volume)
 
     def add(self, object):
-        path = self.volume_location(object)
-        # The last part of the path is the filename for the volume object.
-        volume_filename = path[-1]
-        container = zeit.cms.content.add.find_or_create_folder(*path[:-1])
-        if self._check_duplicate_volume(container, volume_filename):
+        folder, filename = self._create_folder(object, object.product.location)
+        if folder is None:
             return
-        container[volume_filename] = object
-        self._created_object = container[volume_filename]
-        self._finished_add = True
+        folder[filename] = object
+        self._created_object = folder[filename]
+
+        cp_template = zeit.cms.interfaces.ICMSContent(
+            object.product.cp_template, None)
+        if zeit.content.text.interfaces.IPythonScript.providedBy(cp_template):
+            folder, filename = self._create_folder(
+                object, object.product.centerpage)
+            if folder is None:
+                return
```

Comment:
In den Bestandsdaten liegen überall schon `index`-Dateien (benutzen das nur vom XSLT unterstützte `<archive_index>` Format). D.h. der Fall "CP gibt's schon" ist nicht selten. Hach, a pro pos: Das Skript zum Bulk-Anlegen für Ausgabenobjekte müssen wir wohl auch noch erweitern, dass es die dann weglöscht und stattdessen CPs anlegt, oder?

Da wir gesagt haben, wir hängen die Aktion "CP anlegen" an "Volume anlegen" gibt es keine andere Möglichkeit, die CP nachträglich anzulegen (außer Volume wieder löschen...). Insofern ist abbrechen denke ich trotz allem das sinnvollste. Oder wir führen noch irgendwo nen Knopf ein, um die CP anzulegen. Was meinst du?